--- make.include
+++ make.include
@@ -48,7 +48,7 @@ CLINKER	= ${CC} ${OPTFLAGS}
 #CLINKER	= ${CXX} ${OPTFLAGS} -static
 #CLINKER	= link /out:dsdp5.exe

-SH_LD = ${CC} -shared -Wl,-soname,
+SH_LD = ${CC} -shared -Wl,-install_name,

 # STEP 5:
 # Location of BLAS AND LAPACK libraries.
@@ -69,7 +69,7 @@ DSDPMATLABDIR = ${DSDPROOT}/matlab

 # Location of the DSDP library
 DSDPLIB = ${DSDPROOT}/lib/libdsdp.a
-DSDPLIBSO = ${DSDPROOT}/lib/libdsdp.so
+DSDPLIBSO = ${DSDPROOT}/lib/libdsdp.dylib

 # Link drivers to the following libraries.
 # EXAMPLE_LIBS = ${DSDPLIB} ${LAPACKBLAS}
