--- Makefile
+++ Makefile
@@ -54,6 +54,6 @@ oshared:
 	cd tmp; \
 	echo "building ${DSDPLIBSO}"; \
 	${AR} x ${DSDPLIB} ;\
-	${SH_LD} ${DSDPLIBSO} *.o -o ${DSDPLIBSO}; \
+	${SH_LD}${DSDPLIBSO} *.o -o ${DSDPLIBSO} ${LAPACKBLAS}; \
 	cd ../ ; \
 	${RM} tmp
--- make.include
+++ make.include
@@ -43,12 +43,12 @@ DSDPCFLAGS =
 
 
 # STEP 4.  Set Linker and FLAGS:  Link the DSDP library to application
-CFLAGS   = ${OPTFLAGS} -I. -I${DSDPROOT}/include ${DSDPCFLAGS}
+CFLAGS   = ${OPTFLAGS} -I. -I${DSDPROOT}/include -fPIC ${DSDPCFLAGS}
 CLINKER	= ${CC} ${OPTFLAGS}
 #CLINKER	= ${CXX} ${OPTFLAGS} -static
 #CLINKER	= link /out:dsdp5.exe
 
-#SH_LD = ${CC} -shared -W1,-soname,
+SH_LD = ${CC} -shared -Wl,-soname,
 
 # STEP 5:
 # Location of BLAS AND LAPACK libraries. 

